zkevm-circuits/src/state_circuit/state.rs

```diff
@@ -1996,4 +1996,206 @@ mod tests {
             Ok(())
         );
     }
+
+    // To run this benchmark, comment the `ignore` flag and run the following
+    // command:
+    // `RUSTFLAGS='-C target-cpu=native' cargo test --profile bench
+    // bench_state_circuit_prover -- --nocapture`
+    #[ignore]
+    #[test]
+    fn bench_state_circuit_prover() {
```

Comment:
Add `degree` as an arg to the benchmark?

Reply:
That's the nice suggestion!
I am going to add 👍

Reply:
It seems we can pass `degree` as variable because `Circuit` struct parameter need to be defined as const and it comes from `degree`.
https://github.com/NoCtrlZ/zkevm-circuits/blob/evm_state_benches/zkevm-circuits/src/state_circuit/state.rs#L2022

Reply:
@icemelon you mean through env variables? As there's no other way to pass any variables to a test fn.

If so, doesn't seem a bad idea. But we will need `lazy_static` in order to have a const generated from the env_var as the circuit requires it.

@ed255 what do you think?

Reply:
> As there's no other way to pass any variables to a test fn.

Ah I think I can through macro.

Reply:
But instead of making the test more complex with macros. We can simply use and env var and lazy static. Which is much simpler IMO. 

Let's wait for Edu's suggestion

Reply:
I see this:
```rust
        let empty_circuit = StateCircuit::<
            GLOBAL_COUNTER_MAX,
            MEMORY_ROWS_MAX,
            MEMORY_ADDRESS_MAX,
            STACK_ROWS_MAX,
            STACK_ADDRESS_MAX,
            STORAGE_ROWS_MAX,
        >::default();
```
Where some of those generic const parameters are calculated from `DEGREE`.  That means `DEGREE` must be known at compile time.  I think using `lazy_static` with an env_var won't work here, because that would define `DEGREE` at runtime.  Changing this parameter would require a rebuild.

What comes to my mind is to use a `build.rs` that reads the env_var at compile time and then generates a rust file with the parameter.  Or use a macro that reads the env at compile time.  I've found this: https://doc.rust-lang.org/std/macro.env.html which returns an `str`, but we need a `u32`; so we'd need a const fn to parse the `str` to `u32` or something alike.

Reply:
Hey why don't we use macro? 🔥